src/txn/txn_recover.c

```diff
@@ -452,6 +452,69 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the oldest timestamp as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    char *sys_config;
+    const char *sep;
+
+    sys_config = NULL;
+
+    session = r->session;
+    conn = S2C(session);
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ERR_NOTFOUND_OK(
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MIN, &cval), false);
+        if (cval.len != 0)
+            conn->txn_global.snapshot_min = (uint16_t)cval.val;
+
+        WT_ERR_NOTFOUND_OK(
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MAX, &cval), false);
+        if (cval.len != 0) {
```

Comment:
no need for braces here and in the next line also.

Reply:
Fixed